README.md

```diff
@@ -9,16 +9,14 @@
 
 ![STScI Logo](docs/_static/stsci_logo.png)
 
-**JWST requires a C compiler for dependencies and is currently limited to Python 3.9, 3.10 or 3.11.**
+> [!IMPORTANT]
+> JWST requires a C compiler for dependencies and is currently limited to Python 3.10, 3.11, or 3.12.
 
-**Until Python 3.12 is supported, fresh conda environments will require setting the
```

Comment:
Do we actually support py3.12 now (or will support it in the B10.2 release)?  i.e. does everything actually work correctly now under 3.12?

Reply:
I'm not sure if anyone has run the regtests with python 3.12. @zacharyburnett would changing the following be enough to convince jenkins to use python 3.12?
https://github.com/spacetelescope/jwst/blob/e1b07728a2d610e2eae685f36ac9f18905b8e94b/JenkinsfileRT#L46
The unit tests have been running with python 3.12:
https://github.com/spacetelescope/jwst/actions/runs/8345372313/job/22840164182

We have one open PR that is python 3.12 related (which addresses a `DeprecationWarning` introduced in that version):
https://github.com/spacetelescope/jwst/pull/8343

Reply:
yes, that should be enough; I can make a separate PR with that